iree/samples/simple_embedding/BUILD

```diff
@@ -70,112 +70,110 @@ if(${IREE_HAL_DRIVER_DYLIB} AND (${IREE_TARGET_BACKEND_DYLIB-LLVM-AOT} OR DEFINE
-#        "-iree-llvm-debug-symbols=false",
-#    ],
-#)
-#
-#iree_bytecode_module(
-#    name = "simple_embedding_test_bytecode_module_dylib_riscv_64",
-#    src = "simple_embedding_test.mlir",
-#    c_identifier = "iree_samples_simple_embedding_test_module_dylib_riscv_64",
-#    flags = [
-#        "-iree-input-type=mhlo",
-#        "-iree-mlir-to-vm-bytecode-module",
-#        "-iree-hal-target-backends=dylib-llvm-aot",
-#        "-iree-llvm-target-triple=riscv64-pc-linux-elf",
-#        "-iree-llvm-target-cpu=generic-rv64",
-#        "-iree-llvm-target-cpu-features=+m,+a,+f,+d,+c",
-#        "-iree-llvm-target-abi=lp64d",
-#        "-iree-llvm-link-embedded=true",
-#        "-iree-llvm-debug-symbols=false",
-#    ],
-#)
-#
-#iree_bytecode_module(
-#    name = "simple_embedding_test_bytecode_module_dylib_arm_32",
-#    src = "simple_embedding_test.mlir",
-#    c_identifier = "iree_samples_simple_embedding_test_module_dylib_arm_32",
-#    flags = [
-#        "-iree-input-type=mhlo",
-#        "-iree-mlir-to-vm-bytecode-module",
-#        "-iree-hal-target-backends=dylib-llvm-aot",
-#        "-iree-llvm-target-triple=armv7a-pc-linux-elf",
-#        "-iree-llvm-target-float-abi=hard",
-#        "-iree-llvm-link-embedded=true",
-#        "-iree-llvm-debug-symbols=false",
-#    ],
-#)
-#
-#iree_bytecode_module(
-#    name = "simple_embedding_test_bytecode_module_dylib_arm_64",
-#    src = "simple_embedding_test.mlir",
-#    c_identifier = "iree_samples_simple_embedding_test_module_dylib_arm_64",
-#    flags = [
-#        "-iree-input-type=mhlo",
-#        "-iree-mlir-to-vm-bytecode-module",
-#        "-iree-hal-target-backends=dylib-llvm-aot",
-#        "-iree-llvm-target-triple=aarch64-pc-linux-elf",
-#        "-iree-llvm-link-embedded=true",
-#        "-iree-llvm-debug-symbols=false",
-#    ],
-#)
-#
-#run_binary_test(
-#    name = "simple_embedding_embedded_sync_test",
-#    test_binary = ":simple_embedding_embedded_sync",
-#)
+cc_binary(
+    name = "simple_embedding_embedded_sync",
+    srcs = [
+        "device_embedded_sync.c",
+        "simple_embedding.c",
+    ],
+    deps = [
+        ":simple_embedding_test_bytecode_module_dylib_arm_32_c",
+        ":simple_embedding_test_bytecode_module_dylib_arm_64_c",
+        ":simple_embedding_test_bytecode_module_dylib_riscv_32_c",
+        ":simple_embedding_test_bytecode_module_dylib_riscv_64_c",
+        ":simple_embedding_test_bytecode_module_dylib_x86_64_c",
+        "//iree/base",
+        "//iree/hal",
+        "//iree/hal/local",
+        "//iree/hal/local:sync_driver",
+        "//iree/hal/local/loaders:embedded_library_loader",
+        "//iree/modules/hal",
+        "//iree/vm",
+        "//iree/vm:bytecode_module",
+    ],
+)
+
+iree_bytecode_module(
+    name = "simple_embedding_test_bytecode_module_dylib_x86_64",
+    src = "simple_embedding_test.mlir",
+    c_identifier = "iree_samples_simple_embedding_test_module_dylib_x86_64",
+    flags = [
```

Comment:
here's a few more flags (I'll make a rollup at some point so we can just say `-iree-strip-all` or something):

```
-iree-vm-bytecode-module-strip-symbols=true
-iree-vm-bytecode-module-strip-source-map=true
-iree-vm-emit-polyglot-zip=false
```
You lose nice stack traces and such, but that seems fine in this simple_embedding demo and give you smaller binaries!

Reply:
Adding `-iree-vm-bytecode-module-strip-symblos=true` causes execution failure

(at rv32-baremetal)
```
/usr/local/google/home/hcindyl/Projects/iree-dev/iree/iree/vm/bytecode_module.c:166: INVALID_ARGUMENT; mismatched internal_functions/function_descriptors vectors (zu != zu)
```

at x86_64
```
$ ./simple_embedding_dylib 
/usr/local/google/home/hcindyl/Projects/iree-dev/iree/iree/vm/bytecode_module.c:166: INVALID_ARGUMENT; mismatched internal_functions/function_descriptors vectors (0 != 2)
simple_embedding done
```

I will file an issue.

Reply:
Cool! Can fix that, and then this using it will test it :P

Reply:
Flags are added (the problematic one is commented out)